crates/bevy_ecs/src/system/query.rs

```diff
@@ -1129,6 +1129,24 @@ impl<'w, 's, Q: WorldQuery, F: WorldQuery> Query<'w, 's, Q, F> {
     }
 }
 
+impl<'w, 's, Q: WorldQuery, F: WorldQuery> IntoIterator for &'w Query<'_, 's, Q, F> {
+    type Item = ROQueryItem<'w, Q>;
+    type IntoIter = QueryIter<'w, 's, Q, ROQueryFetch<'w, Q>, F>;
+
+    fn into_iter(self) -> Self::IntoIter {
+        self.iter()
+    }
+}
+
+impl<'w, Q: WorldQuery, F: WorldQuery> IntoIterator for &'w mut Query<'_, '_, Q, F> {
+    type Item = QueryItem<'w, Q>;
+    type IntoIter = QueryIter<'w, 'w, Q, QueryFetch<'w, Q>, F>;
```

Comment:
why is this not `QueryIter<'w, 's, ...`

Reply:
I copied the bounds off of iter_mut, since iter_mut bound both 'w and 's to the borrow lifetime i did the same for this impl.

Reply:
alright i guess thats fine then